src/RequestHandler.cpp

```diff
@@ -1,24 +1,72 @@
 #include "RequestHandler.h"
 #include "ConnectKafka.h"
+#include <thread>
+int RequestHandler::Init() {
+  std::string Topic = "MULTIPART_events";
 
-void RequestHandler::PrintToScreen(std::string ToPrint) {
-  std::cout << this->KafkaConnection->GetAllTopics() << std::endl;
-}
-std::string RequestHandler::GetAllTopics() {
-  return this->KafkaConnection->GetAllTopics();
-}
+  std::cout << "_________" << std::endl
+            << KafkaConnection->GetAllTopics() << std::endl
+            << "___________" << std::endl;
 
-bool RequestHandler::CheckIfTopicExists(std::string Topic) {
-  return KafkaConnection->CheckIfTopicExists(Topic);
+  std::vector<std::string> ToSubscribe;
+  if (KafkaConnection->CheckIfTopicExists(Topic)) {
+    ToSubscribe.push_back(Topic);
+    for (auto &SingleStruct : KafkaConnection->GetHighAndLowOffsets(Topic)) {
+      std::cout << SingleStruct.PartitionId << " " << SingleStruct.LowOffset
+                << " " << SingleStruct.HighOffset << std::endl;
+    }
+    SubscribeConsumeAtOffset(Topic, 6000);
+    std::cout << "##################\n##################\n##################\n#"
+                 "#################\n";
+    SubscribeConsumeNLastMessages(Topic, 3);
+  } else
+    std::cout << "No such topic" << std::endl;
+  return 0;
 }
 
-void RequestHandler::Consume(std::string Topic) {
-  KafkaConnection->Consume(Topic);
+std::string RequestHandler::SubscribeConsumeAtOffset(std::string TopicName,
+                                                     int64_t Offset) {
+  int EOFPartitionCounter = 0,
+      NumberOfPartitions =
+          KafkaConnection->GetNumberOfTopicPartitions(TopicName);
```

Comment:
This returns an `int64_t` which you are implicitly casting to `int`, probably best to just use an `int64_t` here.
In other situations if you think the cast is ok then best to make it explicit by using `static_cast<>`.